c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1149,7 +1149,7 @@ class DeviceCachingAllocator {
           "CUDA out of memory. Tried to allocate ",
           format_size(alloc_size),
           ". GPU ",
-          device,
+          std::to_string(device),
```

Comment:
I was confused until I realized that `DeviceIndex` is `int8_t` or simply `char`, so the device index is being interpreted as a character. That's quite the gotcha and I wonder if we should make `DeviceIndex` a proper class to avoid such issues. 

Or if we're settling for the quick workaround, you can avoid creating an extra string with
```suggestion
          static_cast<int>(device),
```